src/FilenameParsing/HashFileIDHelper.php

```diff
@@ -14,70 +13,13 @@
  *
  * e.g.: `Uploads/a1312bc34d/sam__ResizedImageWzYwLDgwXQ.jpg`
  */
-class HashFileIDHelper implements FileIDHelper
+class HashFileIDHelper extends AbstractFileIDHelper
 {
-    use Injectable;
-
     /**
      * Default length at which hashes are truncated.
      */
     const HASH_TRUNCATE_LENGTH = 10;
 
-    public function buildFileID($filename, $hash = null, $variant = null, $cleanfilename = true)
```

Comment:
Moved this method to the abstract class. There were only a very few small places where this implementation differed from `NaturalFileIDHelper`, and those are handled by calling new `protected` methods in the abstract class's implementation of this method.